spotxBidAdapter2.js

```diff
@@ -0,0 +1,362 @@
+        ext.ad_volume = utils.getBidIdParameter('ad_volume', bid.params.video);
+      }
+
+      const mimes = utils.getBidIdParameter('mimes', bid.params.video) != '' ? utils.getBidIdParameter('mimes', bid.params.video) : ['application/javascript', 'video/mp4', 'video/webm'];
+
+      const spotxImp = {
+        id: Date.now(), // Use timestamp as identifier
+        secure: secure,
+        video: {
+          w: contentWidth,
+          h: contentHeight,
+          ext: ext,
+          mimes: mimes
+        }
+      };
+
+      if (utils.getBidIdParameter('price_floor', bid.params) != '') {
+        spotxImp.bidfloor = utils.getBidIdParameter('price_floor', bid.params);
+      }
+
+      if (utils.getBidIdParameter('start_delay', bid.params.video) != '') {
+        spotxImp.startdelay = 0 + Boolean(utils.getBidIdParameter('start_delay', bid.params.video));
+      }
+
+      if (bid.crumbs && bid.crumbs.pubcid) {
+        pubcid = bid.crumbs.pubcid;
+      }
+
+      return spotxImp;
+    });
+    const language = navigator.language ? 'language' : 'userLanguage';
+    const device = {
+      h: screen.height,
+      w: screen.width,
+      dnt: utils.getDNT() ? 1 : 0,
+      language: navigator[language].split('-')[0],
+      make: navigator.vendor ? navigator.vendor : '',
+      ua: navigator.userAgent
+    };
+
+    let requestPayload = {
+      id: channelId,
+      imp: spotxImps,
+      site: {
+        id: siteId,
+        page: page,
+        content: 'content',
+      },
+      device: device,
+      ext: {
+        wrap_response: 1
+      }
+    };
+
+    if (utils.getBidIdParameter('number_of_ads', bid.params)) {
+      requestPayload['ext']['number_of_ads'] = utils.getBidIdParameter('number_of_ads', bid.params);
+    }
+
+    let userExt = {};
+
+    // Add GDPR flag and consent string
+    if (bidderRequest && bidderRequest.gdprConsent) {
+      userExt.consent = bidderRequest.gdprConsent.consentString;
+
+      if (typeof bidderRequest.gdprConsent.gdprApplies === 'boolean') {
+        payload.regs = {
+          ext: {
+            gdpr: (bidderRequest.gdprConsent.gdprApplies ? 1 : 0)
+          }
+        };
+      }
+    }
+
+    // Add common id if available
+    if (pubcid) {
+      userExt.fpc = pubcid;
+    }
+
+    // Only add the user object if it's not empty
+    if (!utils.isEmpty(userExt)) {
+      requestPayload.user = { ext: userExt };
+    }
+
+    return {
+      method: 'POST',
+      url: URL + channelId,
+      data: requestPayload,
+      bidRequest: bidderRequest
+    };
+  },
+
+  /**
+   * Unpack the response from the server into a list of bids.
+   *
+   * @param {*} serverResponse A successful response from the server.
+   * @return {Bid[]} An array of bids which were nested inside the server.
+   */
+  interpretResponse: function(serverResponse, bidderRequest) {
+    const bidResponses = [];
+    serverResponse = serverResponse.body;
```

Comment:
Do we really want to blow away serverResponse here or should we create a new var?